### ISSUE TITLE
params are ignored in production was issue #3111

### ISSUE THREAD

**Issue by geekweapon** [rails/rails#25791]
### Steps to reproduce

Unfortunately I'm working on a private repo at work and I cannot share the entire legacy app codebase but I can provide code snippets.  What is happening is I have a Rails app which has calls and calls belong to a region.  I introduced some params filtering in the view and controller so that I can use a dropdown (form_tag) to filter calls by region instead of all region's calls being shown.  This works fine in development but in a staging/production environment the params are ignored.

### Index action on controller
````
  def index
   if params[:region].present?
     @region = params[:region]
     @assigned = Call.includes(:units, :transferred_from, :transferred_to, :nature, :region, :service_level).where(region_id: params[:region]).assigned_calls.until_end_of_day
     @unassigned = Call.includes(:units, :transferred_from, :transferred_to, :nature, :region, :service_level).where(region_id: params[:region]).unassigned_calls.until_end_of_day
   else
     params[:region] = "1"
     @assigned = Call.includes(:units, :transferred_from, :transferred_to, :nature, :region, :service_level).where(region_id: params[:region]).assigned_calls.until_end_of_day
     @unassigned = Call.includes(:units, :transferred_from, :transferred_to, :nature, :region, :service_level).where(region_id: params[:region]).unassigned_calls.until_end_of_day
   end
    @note = Note.new
    @units = Unit.active.order("unit_name").map{|unit| unit.calls.where(call_status: "open").empty? ? ["#{unit.unit_name} #{unit.unit_type.unit_type} #{unit.status.unit_status}", unit.id] : ["#{unit.unit_name} (on call) #{unit.unit_type.unit_type} #{unit.status.unit_status}", unit.id] }
  end
````

### Index view and partial excerpt
````
div class="page-header well">
  <h3><%= pluralize(@assigned.size, "Active Call") %></h3>
</div>
<%= form_tag calls_path, :method => 'get' do %>
  <%= select_tag "region", options_from_collection_for_select(Region.order(:area), :id, :area, selected: @region), prompt: "Choose Region" %>
  <%= submit_tag "Select", :name => nil, :class => 'btn' %>
<% end %>
<% @assigned.each_with_index do |call, index| %>
  <div class="widget">
    <div class="widget-header">
      <div class="pull-right">

        <%= link_to 'View', call, :class => 'btn btn-primary btn-small'%>
        <% if dispatch? || admin? || manager? || operations? %>
        <%= link_to 'Edit', edit_call_path(call), :class => 'btn btn-info btn-small'%>
        <%= link_to "Close", '#close-modal', data: {toggle: "modal", target: "#close-modal#{index}" }, class: 'btn btn-small btn-warning' %>
        <%= render 'layouts/close_call_modal', call: call, index: index %>
        <%= link_to "Cancel", '#cancel-modal', data: {toggle: "modal", target: "#cancel-modal#{index}" }, class: 'btn btn-small btn-danger' %>
        <%= render 'layouts/cancel_call_modal', call: call, index: index %>
        <%= link_to "Notes", '#note-modal', data: {toggle: "modal", target: "#note-modal#{index}" }, class: 'btn btn-small btn-primary' %>
        <%= render 'layouts/note_modal', call: call, index: index %>
        <% end %>
      </div>
      <i class="icon-phone"></i>
      <h3><%= link_to call.incident_number, call %> <span><%= status(call) %></span></h3>
      <% if call.wait_return == "yes" && call.parent_call_id == nil %>
        <i class="icon-arrow-right dim"></i> <span class="badge badge-important" >Initial Transport</span>
      <% end %>
      <% if call.wait_return == "yes" && call.parent_call_id.present? %>
        <i class="icon-arrow-left dim"></i> <span class="badge badge-important" >Return Transport</span>
      <% end %>
      <% if call.traffic_type == "Emergency" %>
      <span class="badge badge-important"><%= call.traffic_type %></span>
      <% else %>
      <span class="badge badge-info"><%= call.traffic_type %></span>
     <% end %>
     <span class="badge badge-primary" ><%= call.region.try(:area) %></span>
     <span class="badge badge-info" ><%= call.service_level.try(:level_of_service) %></span>
    </div>
<% end %>


### Expected behavior
In development when I select a region from the dropdown/form_tag it passes the region as the param "region" to the controller.  The controller then scopes the calls using a where statement for region_id passing the params[:region] into the controller.  This will allow me to filter the view by regions.  In development this works without an issue.

### Actual behavior
In a production environment (Nginx/Passenger) the params filtering is entirely ignored and it displays all of the calls instead of filtering the params.

Here's a log excerpt of the HTTP GET request made in both dev and production:

#### dev
````
Started GET "/calls?utf8=%E2%9C%93&region=2" for 127.0.0.1 at 2016-07-11 19:39:53 -0500
Processing by CallsController#index as HTML
  Parameters: {"utf8"=>"✓", "region"=>"2"}

#### prod

````
Started GET "/calls?utf8=%E2%9C%93&region=2" for 192.168.130.1 at 2016-07-11 19:39:17 -0500
Processing by CallsController#index as HTML
  Parameters: {"utf8"=>"✓", "region"=>"2"}
````

#### System configuration
**Rails version**: 3.2.22

**Ruby version**: 1.9.3

**Comment by matthewd** [comment]
Rails 3.2 is no longer supported.

**Comment by geekweapon** [comment]
@matthewd Thanks for the heads up, but do you have any insight on what could be causing this?  I'm not in a position to upgrade an entire production app to 4.2 just yet without a gigantic overhaul.

**Comment by geekweapon** [comment]
@matthewd Fixed the problem and it's definitely a 3.2.x bug.  The only way I can get it working in production is to turn the log level to debug in production.  Works like a charm but is nasty.  I'm probably going to backport a patch in case anyone wants a PR against this series.  My next step is to upgrade to 4.2.6.
